pkg/cmd/operator/webhooks_test.go

```diff
@@ -326,3 +336,273 @@ func createFileWithContent(path string, data []byte) error {
+					},
+					ValidateUpdateFunc: func(_, _ *corev1.Pod) field.ErrorList {
+						return nil
+					},
+					GetWarningsOnCreateFunc: func(_ *corev1.Pod) []string {
+						panic("unexpected call to GetWarningsOnCreateFunc")
+					},
+					GetWarningsOnUpdateFunc: func(_, _ *corev1.Pod) []string {
+						return nil
+					},
+				},
+			},
+			expectedWarnings: nil,
+			expectedError:    nil,
+		},
+		{
+			name: "update, error, no warnings",
+			ar: func() *admissionv1.AdmissionReview {
+				return &admissionv1.AdmissionReview{
+					Request: &admissionv1.AdmissionRequest{
+						UID: "uid",
+						Resource: metav1.GroupVersionResource{
+							Group:    "",
+							Version:  "v1",
+							Resource: "pods",
+						},
+						Operation: "UPDATE",
+						Object:    runtime.RawExtension{Raw: podRaw},
+						OldObject: runtime.RawExtension{Raw: podRaw},
+					},
+				}
+			}(),
+			validators: map[schema.GroupVersionResource]Validator{
+				corev1.SchemeGroupVersion.WithResource("pods"): &GenericValidator[*corev1.Pod]{
+					ValidateCreateFunc: func(_ *corev1.Pod) field.ErrorList {
+						panic("unexpected call to ValidateCreateFunc")
+					},
+					ValidateUpdateFunc: func(_, _ *corev1.Pod) field.ErrorList {
+						return field.ErrorList{
+							field.Invalid(field.NewPath("spec"), "value", "error"),
+						}
+					},
+					GetWarningsOnCreateFunc: func(_ *corev1.Pod) []string {
+						panic("unexpected call to GetWarningsOnCreateFunc")
+					},
+					GetWarningsOnUpdateFunc: func(_, _ *corev1.Pod) []string {
+						return nil
+					},
+				},
+			},
+			expectedWarnings: nil,
+			expectedError: apierrors.NewInvalid(corev1.SchemeGroupVersion.WithKind("Pod").GroupKind(), "pod", field.ErrorList{
+				field.Invalid(field.NewPath("spec"), "value", "error"),
+			}),
+		},
+		{
+			name: "update, no errors, warnings",
+			ar: func() *admissionv1.AdmissionReview {
+				return &admissionv1.AdmissionReview{
+					Request: &admissionv1.AdmissionRequest{
+						UID: "uid",
+						Resource: metav1.GroupVersionResource{
+							Group:    "",
+							Version:  "v1",
+							Resource: "pods",
+						},
+						Operation: "UPDATE",
+						Object:    runtime.RawExtension{Raw: podRaw},
+						OldObject: runtime.RawExtension{Raw: podRaw},
+					},
+				}
+			}(),
+			validators: map[schema.GroupVersionResource]Validator{
+				corev1.SchemeGroupVersion.WithResource("pods"): &GenericValidator[*corev1.Pod]{
+					ValidateCreateFunc: func(_ *corev1.Pod) field.ErrorList {
+						panic("unexpected call to ValidateCreateFunc")
+					},
+					ValidateUpdateFunc: func(_, _ *corev1.Pod) field.ErrorList {
+						return nil
+					},
+					GetWarningsOnCreateFunc: func(_ *corev1.Pod) []string {
+						panic("unexpected call to GetWarningsOnCreateFunc")
+					},
+					GetWarningsOnUpdateFunc: func(_, _ *corev1.Pod) []string {
+						return []string{"warning"}
+					},
+				},
+			},
+			expectedWarnings: []string{"warning"},
+			expectedError:    nil,
+		},
```

Comment:
Please consider adding `update, errors, warnings` and `create, errors, warnings` cases as well.